src/driver/opam.ml

```diff
@@ -135,6 +135,10 @@ let pkg_to_dir_map () =
               | "doc" :: _pkg :: "odoc-pages" :: _ ->
                   Logs.debug (fun m -> m "Found odoc page: %a" Fpath.pp fpath);
 
+                  (Fpath.Set.add Fpath.(v prefix // fpath) odoc_pages, others)
+              | "doc" :: _pkg :: "odoc-assets" :: _ ->
```

Comment:
Unrelated to this PR but pages and assets are taken from the list of installed files for a package. I find that unnecessary and potentially buggy. A package could install doc in an other package or this file could be outdated (`dune install`), which might crash the driver.
I think the `doc/pkgname` folder should be scanned independently of `pkg_contents`. I think the same for libraries but there might be cases that require this?